crates/libs/sys/src/core/mod.rs

```diff
@@ -35,6 +35,29 @@ impl GUID {
     }
 }
 
+#[cfg(all(windows_raw_dylib, target_arch = "x86"))]
+#[macro_export]
+macro_rules! link {
+    ($library:literal $abi:literal $(#[$($doc:tt)*])* fn $name:ident($($arg:ident: $argty:ty),*)->$ret:ty) => (
+        #[link(name = $library, kind = "raw-dylib", modifiers = "+verbatim", import_name_type = "undecorated")]
+        extern $abi {
+            pub fn $name($($arg: $argty),*) -> $ret;
+        }
+    )
+}
+
+#[cfg(all(windows_raw_dylib, not(target_arch = "x86")))]
+#[macro_export]
+macro_rules! link {
+    ($library:literal $abi:literal $(#[$($doc:tt)*])* fn $name:ident($($arg:ident: $argty:ty),*)->$ret:ty) => (
+        #[link(name = $library, kind = "raw-dylib", modifiers = "+verbatim")]
+        extern "system" {
```

Comment:
Is there a specific reason this was made to ignore $abi? (it's now "C" rather than "system", as a workaround for variadic functions).

Reply:
Yes, it's a workaround for this: https://github.com/rust-lang/rust/issues/110505

Reply:
Yes, that's why it's C now, but why was it "system" rather than $abi?

Reply:
Because the ABI is generally only relevant for x86 and "system" should work for all ARM64 and x64 Windows APIs.